dejavu_bot.py

```diff
@@ -116,35 +157,69 @@ async def create_and_send_image(text, channel):
 
     # Save the image to a BytesIO buffer
     buffer = BytesIO()
-    img.save(buffer, 'png')
+    img.save(buffer, "png")
     buffer.seek(0)  # Reset the buffer pointer to the beginning
 
-    file = discord.File(buffer, filename='image.png')
+    file = discord.File(buffer, filename="image.png")
     await channel.send(file=file)
 
-async def who_said(who_said_content, channel):
+
+async def whosaid(message, channel):
     """
-    Ask who said who_said_content and store the context
-    in the bot's who_said_context
+    A game where a message is presented and the user has to guess who wrote it
+    by mentioning the user. They get 2 guesses before game over.
+    /dejavu whosaid
     """
-    await channel.send('Who said: ' + who_said_content)
+
+    # Set inital game variables and start the game
+    bot.whosaid["playing"] = True
+    bot.whosaid["channel"] = message.channel.id
+    bot.whosaid["second_chance"] = True
+    bot.whosaid["author"] = message.author.name
+    await channel.send("Who said: " + message.content)
+
 
 @bot.event
 async def on_message(message):
     """
-    check if the id in the response matches the
-    id if the whosaid game is being played
+    Check if the ID in the response matches the
+    ID of the user whose message got fetched by
+    the whosaid game and make sure we're in the
+    channel the game started in.
+
+    Then, the game logic runs.
     """
-    if message.author == bot.user:
+
+    if message.author.bot == True:
+        return
+    elif bot.whosaid["channel"] != message.channel.id:
         return
```

Comment:
These can be combined.

```suggestion
    if message.author.bot == True or bot.whosaid["channel"] != message.channel.id:
        return
```

Reply:
:+1: